internal/pkg/term/progress/ecs.go

```diff
@@ -0,0 +1,157 @@
+	done   chan struct{}            // Channel that's closed when there are no more events to listen on.
+	mu     sync.Mutex               // Lock used to mutate data to render.
+}
+
+// Listen updates the deployment statuses and failure event messages as events are streamed.
+func (c *rollingUpdateComponent) Listen() {
+	for ev := range c.stream {
+		c.mu.Lock()
+		c.deployments = ev.Deployments
+		c.failureMsgs = append(c.failureMsgs, ev.LatestFailureEvents...)
+		if len(c.failureMsgs) > c.maxLenFailureMsgs {
+			c.failureMsgs = c.failureMsgs[len(c.failureMsgs)-c.maxLenFailureMsgs:]
+		}
+		c.mu.Unlock()
+	}
+	close(c.done)
+}
+
+// Render prints first the deployments as a tableComponent and then the failure messages as singleLineComponents.
+func (c *rollingUpdateComponent) Render(out io.Writer) (numLines int, err error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	buf := new(bytes.Buffer)
+
+	nl, err := c.renderDeployments(buf)
+	if err != nil {
+		return 0, err
+	}
+	numLines += nl
+
+	nl, err = c.renderFailureMsgs(buf)
+	if err != nil {
+		return 0, err
+	}
+	numLines += nl
+
+	if _, err := buf.WriteTo(out); err != nil {
+		return 0, fmt.Errorf("render rolling update component to writer: %w", err)
+	}
+	return numLines, nil
+}
+
+// Done returns a channel that's closed when there are no more events to listen.
+func (c *rollingUpdateComponent) Done() <-chan struct{} {
+	return c.done
+}
+
+func (c *rollingUpdateComponent) renderDeployments(out io.Writer) (numLines int, err error) {
+	header := []string{"", "Revision", "Rollout", "Desired", "Running", "Failed", "Pending"}
+	var rows [][]string
+	for _, d := range c.deployments {
+		rows = append(rows, []string{
+			d.Status,
+			d.TaskDefRevision,
+			prettifyRolloutStatus(d.RolloutState),
+			strconv.Itoa(d.DesiredCount),
+			strconv.Itoa(d.RunningCount),
+			strconv.Itoa(d.FailedCount),
+			strconv.Itoa(d.PendingCount),
+		})
+	}
+	table := newTableComponent("Deployments", header, rows)
+	table.Padding = c.padding
+	nl, err := table.Render(out)
+	if err != nil {
+		return 0, fmt.Errorf("render deployments table: %w", err)
+	}
+	return nl, err
+}
+
+func (c *rollingUpdateComponent) renderFailureMsgs(out io.Writer) (numLines int, err error) {
+	if len(c.failureMsgs) == 0 {
+		return 0, nil
+	}
+
+	title := "Latest failure event"
+	if l := len(c.failureMsgs); l > 1 {
+		title = fmt.Sprintf("Latest %d failure events", l)
+	}
+	components := []Renderer{
+		&singleLineComponent{
+			Text:    title,
+			Padding: c.padding,
+		},
+	}
+
+	for _, msg := range reverseStrings(c.failureMsgs) {
+		for i, truncatedMsg := range splitByLength(msg, maxCellLength) {
+			pretty := fmt.Sprintf("  %s", truncatedMsg)
+			if i == 0 {
+				pretty = fmt.Sprintf("- %s", truncatedMsg)
+			}
+			components = append(components, &singleLineComponent{
+				Text:    pretty,
+				Padding: c.padding + nestedComponentPadding,
+			})
+		}
+	}
+	return renderComponents(out, components)
```

Comment:
Do we need `return renderComponents(out, components) + 1`, counting the title?

Reply:
The title is a member of the `components` slice, so it will be counted :D